packages/app-utils/src/components/ConnectionBootstrap.tsx

```diff
@@ -115,6 +118,8 @@ export function ConnectionBootstrap({
         log.info('Shutdown', `${JSON.stringify(detail)}`);
         setError(`Server shutdown: ${detail ?? 'Unknown reason'}`);
         setIsShutdown(true);
+        setIsReconnecting(false);
+        setIsAuthFailed(false);
```

Comment:
You could have a string represent the connection state instead. Since it should only be 1 of the states at a time. This would reduce the needed to set multiple states any time the connection state changes.

```ts
const [connectionState, setConnectionState] = useState<'connected' | 'connecting' | 'reconnecting' | 'failed' | 'shutdown'>('connecting');

const isAuthFailed = connectionState === 'failed';
const isShutdown = connectionState === 'shutdown';
```

The `isAuthFailed` is cheap to calculate here, so deriving the state is a good way to keep the change of state simpler here.